src/components/interface/cbuf_mgr/cbuf_meta.h

```diff
@@ -0,0 +1,139 @@
+#define CBUFM_NEXT_MASK 0xf
+#define CBUFM_PTR_MASK 0xfff
+#define CBUFM_INC_UNIT     (1<<CBUFM_FIRST_FLAGS_SZ)
+#define CBUFM_DEC_UNIT     (~(CBUFM_INC_UNIT)+1)
+#define CBUF_REFCNT_MAX   ((1<<CBUFM_REFCNT_SZ)-1)
+#define CBUF_SENDRECV_MAX  ((1<<8)-1)
+
+/*get flags*/
+#define CBUF_OWNER(meta)  ((meta)->nfo & CBUFM_OWNER)
+#define CBUF_TMEM(meta) ((meta)->nfo & CBUFM_TMEM)
+#define CBUF_INCONSISENT(meta) ((meta)->nfo & CBUFM_INCONSISENT)
+#define CBUF_WRITABLE(meta) ((int)((meta)->next_flag) & CBUFM_WRITABLE)
+#define CBUF_TOUCHED(meta) ((int)((meta)->next_flag) & CBUFM_TOUCHED)
+#define CBUF_RELINQ(meta) ((int)((meta)->next_flag) & CBUFM_RELINQ)
+/*set flags*/
+#define CBUF_SET_OWNER(meta) (meta)->nfo |= CBUFM_OWNER
+#define CBUF_SET_TMEM(meta) (meta)->nfo |= CBUFM_TMEM
+#define CBUF_SET_INCONSISENT(meta) (meta)->nfo |= CBUFM_INCONSISENT
+#define CBUF_SET_WRITABLE(meta) (meta)->next_flag = (struct cbuf_meta *)((int)(meta)->next_flag | CBUFM_WRITABLE)
+#define CBUF_SET_TOUCHED(meta) (meta)->next_flag = (struct cbuf_meta *)((int)(meta)->next_flag | CBUFM_TOUCHED)
+#define CBUF_SET_RELINQ(meta) (meta)->next_flag = (struct cbuf_meta *)((int)(meta)->next_flag | CBUFM_RELINQ)
+/*unset flags*/
+#define CBUF_UNSET_OWNER(meta) (meta)->nfo &= (~CBUFM_OWNER)
+#define CBUF_UNSET_TMEM(meta) (meta)->nfo &= (~CBUFM_TMEM)
+#define CBUF_UNSET_INCONSISENT(meta) (meta)->nfo = (int)(meta)->nfo & (~CBUFM_INCONSISENT)
+#define CBUF_UNSET_WRITABLE(meta) (meta)->next_flag = (struct cbuf_meta *)((int)(meta)->next_flag & (~CBUFM_WRITABLE))
+#define CBUF_UNSET_TOUCHED(meta) (meta)->next_flag = (struct cbuf_meta *)((int)(meta)->next_flag & (~CBUFM_TOUCHED))
+#define CBUF_UNSET_RELINQ(meta) (meta)->next_flag = (struct cbuf_meta *)((int)(meta)->next_flag & (~CBUFM_RELINQ))
+
+
+#define CBUFM_GET_PTR(meta) (((meta)->nfo)>>12)
+#define CBUFM_SET_PTR(meta, v) do {         \
+	(meta)->nfo &= CBUFM_PTR_MASK;      \
+	(meta)->nfo |= (v); } while(0)        
+#define CBUFM_GET_REFCNT(meta) (((meta)->nfo>>CBUFM_FIRST_FLAGS_SZ) & 0x1ff)
+#define CBUFM_INC_REFCNT(meta) cos_faa((int *)(&((meta)->nfo)), CBUFM_INC_UNIT)
+#define CBUFM_DEC_REFCNT(meta) cos_faa((int *)(&((meta)->nfo)), CBUFM_DEC_UNIT)
+#define CBUFM_GET_NEXT(meta) ((struct cbuf_meta *)((int)((meta)->next_flag) & ~CBUFM_NEXT_MASK))
+#define CBUFM_SET_NEXT(meta, next) do {                \
+	(meta)->next_flag = (struct cbuf_meta *)((int)((meta)->next_flag) & CBUFM_NEXT_MASK);   \
+	(meta)->next_flag = (struct cbuf_meta *)((int)((meta)->next_flag) | (int)(next)); } while(0)
+#define CBUFM_SET_NEXT_NULL(meta) (meta)->next_flag = (struct cbuf_meta *)((int)((meta)->next_flag) & CBUFM_NEXT_MASK)
+
+/** 
+ * manually assign each bit to prevent compiler reorder   
+ *                                                        
+ * structure of nfo in meta data as follows:              
+ * +-------20-----+-----9-----+---------3---------------+ 
+ * | page pointer |  refcnt   | owner|tmem|inconsistent | 
```

Comment:
You can shrink the `refcnt` to 8 bits, and move the `relinq` bit to this flag field so that you can get rid of the other flags.  We don't need `writable` or `touched`.  The `next_flag` struct can be devoted to only the next pointer.